heron/schedulers/src/java/com/twitter/heron/scheduler/kubernetes/KubernetesConstants.java

```diff
@@ -74,10 +74,18 @@ private KubernetesConstants() {
   public static final String METRICS_CACHE_STATS_PORT = "6008";
   public static final String CHECKPOINT_MGR_PORT = "6009";
 
-  public static final String[] PORT_LIST = new String[]{
-      MASTER_PORT, TMASTER_CONTROLLER_PORT, TMASTER_STATS_PORT,
-      SHELL_PORT, METRICSMGR_PORT, SCHEDULER_PORT, METRICS_CACHE_MASTER_PORT,
-      METRICS_CACHE_STATS_PORT, CHECKPOINT_MGR_PORT };
+  public static final Map<ExecutorPort, String> EXECUTOR_PORTS = new HashMap<>();
+  static {
+    EXECUTOR_PORTS.put(ExecutorPort.MASTER_PORT, MASTER_PORT);
```

Comment:
I suggested a method `getRequiredPorts` in one of the comments. Please see if using that would be better. If a required port is not `put` into the `EXECUTOR_PORTS` throw an error.

Reply:
Please see me previous comment concerning the current implementation will already throw an exception if a required port is missing.

Reply:
I will replace getNumOfRequiredPorts with getRequiredPorts as it is a more generic method